app/controllers/api/v1/users_controller.rb

```diff
@@ -70,14 +84,23 @@ def search_user_and_render_json
         records = @users.search({
                     search_text: params['searchText'],
                     role_name: params['role_name']}).limit(25)
-        data = ActiveModel::ArraySerializer.new(records, each_serializer: serializer, root: "users").as_json
+        data = ActiveModel::ArraySerializer.new(records,
+          each_serializer: serializer,
+          include_user_roles: true,
+          root: "users").as_json
         render json: { "meta": {"search": params["searchText"] } }.merge(data)
       end
 
+      def can_update_other_user?
+        User.current_user.can_manage_users? &&
+        User.current_user.id != params["id"].to_i
+      end
+
       def user_params
-        attributes = %i[last_connected last_disconnected
-        first_name last_name email receive_email other_phone title mobile printer_id]
-        attributes.concat([:user_role_ids]) if User.current_user.supervisor?
+        attributes = %i[image_id first_name last_name email receive_email
+          other_phone title mobile printer_id]
+        attributes.concat([:disabled]) if can_update_other_user?
```

Comment:
We need to guard the setting of the disabled attribute. Can you create a 'can_disable_users' permission and assign to System administrators.